src/main/java/org/mastodon/mamut/fitting/ellipsoid/FitEllipsoid.java

```diff
@@ -83,6 +86,12 @@ public static Ellipsoid yuryPetrov( final double[][] points )
 
 		final double[][] DTD = new double[ 9 ][ 9 ];
 		LinAlgHelpers.multATB( d, d, DTD );
+		CholeskyDecomposition choleskyDecomposition = new CholeskyDecomposition( new Matrix( DTD ) );
+
+		// symmetric, positive, definite?
+		if ( !choleskyDecomposition.isSPD() )
+			// in this case, ellipsoid fitting is not possible
+			return null;
 		final Matrix V = new CholeskyDecomposition( new Matrix( DTD ) ).solve( new Matrix( b ) );
```

Comment:
The code calls ```new CholeskyDecomposition( new Matrix( DTD ) )``` twice. Is there a reason for that. I think calling it once would be enough.